File: doc-examples/example-java/src/test/java/example/BusinessRulesTest.java
package example;

import io.micronaut.context.ApplicationContext;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

class BusinessRulesTest {

    @Test
    void theRulesGuardTheMethods() {
        try (ApplicationContext context = ApplicationContext.run()) {
            BusinessRules rules = context.getBean(BusinessRules.class);

            assertEquals("registered Ann", rules.register(new Customer("Ann", 34, "CZ")));
            NotEligibleException rejected = assertThrows(NotEligibleException.class,
                () -> rules.register(new Customer("Bob", 15, "CZ")));
            assertEquals("Bob must be an adult in Europe", rejected.getMessage());

            assertEquals("deposited 101", rules.deposit(101));
            NotEligibleException tooSmall = assertThrows(NotEligibleException.class, () -> rules.deposit(100));
            assertEquals("Must be greater than 100", tooSmall.getMessage());
        }
    }
}

File: doc-examples/example-kotlin/src/test/kotlin/example/BusinessRulesTest.kt
package example

import io.micronaut.context.ApplicationContext
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test

class BusinessRulesTest {

    @Test
    fun theRulesGuardTheMethods() {
        ApplicationContext.run().use { context ->
            val rules = context.getBean(BusinessRules::class.java)

            assertEquals("registered Ann", rules.register(Customer("Ann", 34, "CZ")))
            val rejected = assertThrows(NotEligibleException::class.java) {
                rules.register(Customer("Bob", 15, "CZ"))
            }
            assertEquals("Bob must be an adult in Europe", rejected.message)

            assertEquals("deposited 101", rules.deposit(101))
            val tooSmall = assertThrows(NotEligibleException::class.java) { rules.deposit(100) }
            assertEquals("Must be greater than 100", tooSmall.message)
        }
    }
}
